modules/fedimint-walletv2-client/src/lib.rs

```diff
@@ -0,0 +1,570 @@
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct WalletClientInit {
+    pub esplora_connection: DynEsploraConnection,
+}
+
+impl Default for WalletClientInit {
+    fn default() -> Self {
+        Self {
+            esplora_connection: DynEsploraConnection::from(RealEsploraConnection),
+        }
+    }
+}
+
+impl ModuleInit for WalletClientInit {
+    type Common = WalletCommonInit;
+
+    async fn dump_database(
+        &self,
+        _dbtx: &mut DatabaseTransaction<'_>,
+        _prefix_names: Vec<String>,
+    ) -> Box<dyn Iterator<Item = (String, Box<dyn erased_serde::Serialize + Send>)> + '_> {
+        Box::new(BTreeMap::new().into_iter())
+    }
+}
+
+#[apply(async_trait_maybe_send!)]
+impl ClientModuleInit for WalletClientInit {
+    type Module = WalletClientModule;
+
+    fn supported_api_versions(&self) -> MultiApiVersion {
+        MultiApiVersion::try_from_iter([ApiVersion { major: 0, minor: 0 }])
+            .expect("no version conflicts")
+    }
+
+    async fn init(&self, args: &ClientModuleInitArgs<Self>) -> anyhow::Result<Self::Module> {
+        Ok(WalletClientModule {
+            root_secret: args.module_root_secret().clone(),
+            cfg: args.cfg().clone(),
+            client_ctx: args.context(),
+            db: args.db().clone(),
+            module_api: args.module_api().clone(),
+            esplora_rpc: self.esplora_connection.clone(),
+        })
+    }
+
+    fn get_database_migrations(&self) -> BTreeMap<DatabaseVersion, ClientMigrationFn> {
+        BTreeMap::new()
+    }
+}
+
+impl WalletClientModule {
+    /// Fetch the total value of bitcoin controlled by the federation.
+    pub async fn federation_value(&self) -> FederationResult<bitcoin::Amount> {
+        self.module_api
+            .federation_wallet()
+            .await
+            .map(|tx_out| tx_out.map_or(bitcoin::Amount::ZERO, |tx_out| tx_out.value))
+    }
+
+    /// Fetch the consensus block count of the federation which trails the
+    /// bitcoin block count by at least six blocks.
+    pub async fn consensus_block_count(&self) -> FederationResult<u64> {
+        self.module_api.consensus_block_count().await
+    }
+
+    /// Fetch the set of bitcoin transactions generated by the federation that
+    /// are currently still pending.
+    pub async fn pending_transactions(&self) -> FederationResult<BTreeSet<bitcoin::Txid>> {
+        self.module_api.pending_transactions().await
+    }
+
+    /// Fetch the current fee required to send an on-chain payment.
+    pub async fn send_fee(&self) -> Result<bitcoin::Amount, SendError> {
+        self.module_api
+            .send_fee()
+            .await
+            .map_err(|e| SendError::FederationError(e.to_string()))?
+            .ok_or(SendError::NoConsensusFeerateAvailable)
+            .map(|fee| fee.value)
+    }
+
+    /// Send an on-chain payment with the given fee.
+    pub async fn send(
+        &self,
+        address: &Address<NetworkUnchecked>,
```

Comment:
Conceptually we should be passing `Address` (which is the same as `Address<NetworkChecked>` around. We should only need to deal with `NetworkUnchecked` on the boundaries of our system, check them fast and not have to worry about it anymore. Ping @tvolk131 if that's our approach, or are there any things preventing it.

Reply:
Hmm... after noticing the check just below. I guess it makes sense to validate it here, since this is a public method that a downstream might use. Though... maybe we should pass that responsibilty to the user (that's what using `Address<NetworkChecked>` would do). Possibly they already validated it anyway, and they already have it checked.